core/domain/html_cleaner_test.py

```diff
@@ -719,7 +719,7 @@ def test_convert_to_ckeditor(self):
                 '&amp;amp;amp;quot;\\&amp;quot; filepath-with-value=\\&amp;quot'
                 ';&amp;amp;amp;quot;loremipsum.png&amp;amp;amp;quot;\\&amp;quot'
                 ';&amp;gt;&amp;lt;/oppia-noninteractive-image&amp;gt;&amp;lt;'
-                'p&amp;gt;&amp;amp;amp;nbsp;&amp;lt;/p&amp;gt;&amp;lt;/li&amp;'
+                'p&amp;gt;\\u00a0&amp;lt;/p&amp;gt;&amp;lt;/li&amp;'
```

Comment:
Is there a reason not to use `formatter='html'` as in the SO post linked above? Without that, you get the actual unicode character (instead of `&nbsp;`).

Functionally, I see no difference in the browser, just wondering why not use `formatter='html'`.

Reply:
So, when loaded and re-saved, is the saved string still the unicode character?

(You can also check this more robustly by making a change to the non-affected part of the string.)

Reply:
(Also note that it would be a good idea to add a backend test for this.)

Reply:
So what I did is I added `<p><br></p>` in an exploration loaded that and checked in inspect element - it shows `<p>&nbsp;</p>`. I downloaded the exploration through history tab and it replaced this line with `<p>\_</p>`. 

Then I created an exploration using the editor (CKEditor in oppia) pressed enter twice to create an empty line. This too had same content in inspect element `<p>&nbsp;</p>`. Downloaded this exploration and checked that it too was changed to `<p>\_</p>`.

So I think it's good!